browser/brave_content_browser_client_browsertest.cc

```diff
@@ -716,3 +723,88 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, CheckExpectedExtensions) {
   // Brave Extension background page should be disabled by default.
   EXPECT_FALSE(extensions::BackgroundInfo::HasBackgroundPage(brave_extension));
 }
+
+#if BUILDFLAG(IS_WIN)
+
+class WinBraveContentBrowserClientTest
+    : public BraveContentBrowserClientTest,
+      public ::testing::WithParamInterface<
+          base::test::ScopedOSInfoOverride::Type> {
+ public:
+  WinBraveContentBrowserClientTest() = default;
+  ~WinBraveContentBrowserClientTest() override = default;
+
+  bool GetShouldDoLearning(Browser* browser) {
+    return browser->tab_strip_model()
+        ->GetActiveWebContents()
+        ->GetShouldDoLearningForTesting();
+  }
+
+  bool IsWin11() const {
+    return base::win::GetVersion() >= base::win::Version::WIN11;
+  }
+
+ private:
+  base::test::ScopedOSInfoOverride win_version_{GetParam()};
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    WinBraveContentBrowserClientTest,
+    ::testing::Values(base::test::ScopedOSInfoOverride::Type::kWin11Home,
+                      base::test::ScopedOSInfoOverride::Type::kWin10Home));
+
+IN_PROC_BROWSER_TEST_P(WinBraveContentBrowserClientTest, PRE_WindowsRecall) {
+  EXPECT_EQ(windows_recall::IsWindowsRecallAvailable(), IsWin11());
+
+  auto* local_state = g_browser_process->local_state();
+
+  if (IsWin11()) {
+    EXPECT_TRUE(
+        local_state->GetBoolean(windows_recall::prefs::kBlockWindowsRecall));
+    EXPECT_TRUE(windows_recall::IsWindowsRecallDisabled(local_state));
```

Comment:
this assertion has side effects that can affect the rest of the test because calling `IsWindowsRecallDisabled` does a lazy cache of the value